test/services/sync/strategies/consumerActions/UserAction.test.js

```diff
@@ -234,4 +237,121 @@ describe('User Actions', () => {
 			});
 		});
 	});
+	describe('autoMatchImportUser', () => {
+		let findBySchoolAndNameStub;
+		let findImportUsersBySchoolAndNameStub;
+		let createOrUpdateImportUserStub;
+
+		beforeEach(() => {
+			findBySchoolAndNameStub = sinon.stub(UserRepo, 'findBySchoolAndName');
+			findImportUsersBySchoolAndNameStub = sinon.stub(UserRepo, 'findImportUsersBySchoolAndName');
+			createOrUpdateImportUserStub = sinon.stub(UserRepo, 'createOrUpdateImportUser');
+		});
+
+		it('should not create match, when firstName or lastName are empty', async () => {
+			const schoolId = 'foo';
+			const userUpdateObject1 = {
+				firstName: ' ',
+				lastName: 'doe',
+			};
+			const userUpdateObject2 = {
+				firstName: 'john',
+				lastName: '',
+			};
+
+			await userAction.autoMatchImportUser(schoolId, userUpdateObject1);
+			expect(userUpdateObject1.match).to.be.undefined;
+
+			await userAction.autoMatchImportUser(schoolId, userUpdateObject2);
+			expect(userUpdateObject2.match).to.be.undefined;
+		});
+
+		it('should not create match, when no local user is found', async () => {
+			const schoolId = 'foo';
+			const userUpdateObject = {
+				firstName: 'john',
+				lastName: 'doe',
+			};
+			findBySchoolAndNameStub.resolves([]);
+			await userAction.autoMatchImportUser(schoolId, userUpdateObject);
+			expect(userUpdateObject.match).to.be.undefined;
+		});
+
+		it('should not create match, when multiple local user are found', async () => {
+			const schoolId = 'foo';
+			const userUpdateObject = {
+				firstName: 'john',
+				lastName: 'doe',
+			};
+			findBySchoolAndNameStub.resolves(['dummyUser1', 'dummyUser2', 'dummyUser3']);
+			await userAction.autoMatchImportUser(schoolId, userUpdateObject);
+			expect(userUpdateObject.match).to.be.undefined;
+		});
+
+		it('should create match, when only 1 local user is found', async () => {
+			const schoolId = 'foo';
+			const userUpdateObject = {
+				firstName: 'john',
+				lastName: 'doe',
+			};
+			const localUserId = 'bar';
+			findBySchoolAndNameStub.resolves([{ _id: localUserId }]);
+			findImportUsersBySchoolAndNameStub.resolves([]);
+
+			await userAction.autoMatchImportUser(schoolId, userUpdateObject);
+			expect(userUpdateObject.match.userId).to.equal(localUserId);
+			expect(userUpdateObject.match.matchedBy).to.equal('auto');
+		});
+
+		it('should not create match, when multiple import users with same name exists', async () => {
+			const schoolId = 'foo';
+			const userUpdateObject = {
+				firstName: 'john',
+				lastName: 'doe',
+			};
+			const localUserId = 'bar';
+			findBySchoolAndNameStub.resolves([{ _id: localUserId }]);
+			findImportUsersBySchoolAndNameStub.resolves(['dummyImportUser']);
+
+			await userAction.autoMatchImportUser(schoolId, userUpdateObject);
+			expect(userUpdateObject.match).to.be.undefined;
+		});
+
+		it('should revoke any previously matched import user, when they have the same name', async () => {
```

Comment:
```suggestion
		it('should revoke any previously auto-matched import user, when they have the same name', async () => {
```
it should apply only for auto matches